external/avro/src/main/scala/org/apache/spark/sql/v2/avro/AvroScanBuilder.scala

```diff
@@ -50,7 +52,7 @@ class AvroScanBuilder (
     if (sparkSession.sessionState.conf.avroFilterPushDown) {
```

Comment:
I think this is a bug in file source v2. This conf should only affect avro filter pushdown, but not partition pruning (you can check file source v1).

The fix should be
```
abstract class FileScanBuilder ... with SupportsPushDownFilters {
  ...
  protected var partitionFilters = Seq.empty[Filter]
  protected var dataFilters = Seq.empty[Filter]
  override def pushFilters(filters: Array[Filter]): Array[Filter] = {
    // split the filters
    pushDataFilters()
  }
  // Overwrite it if source supports filter pushdown
  protected def pushDataFilters(): Seq[Filter] = dataFilters
}
```